### ISSUE TITLE
Question on the Velocity in the X and Y Axes of the Ground Truth

### ISSUE THREAD

**Issue by hoangvietdo** [StironjaVlaho/RAVE#1]
Hi,

I have a question regarding the ground truth (GT) velocity of the "Mocap dark fast" dataset. Specifically, I attempted to plot the GT velocity, but the values on the x and y axes do not match those in Fig. 3 of your paper.

Could you please explain how you obtained these values?

**Comment by StironjaVlaho** [comment]
Hi,
first of all, thank you very much for your interest in our work. 

We transformed ground truth velocity in the radar's body frame using Equation 3 from our paper. As the translational distance between frames is negligible within the IRS dataset, we opted to neglect the second part of the equation (omega x p). Therefore, it is necessary to transform ground-truth velocity values into radar's body frame.

**Comment by hoangvietdo** [comment]
Thanks for your fast response.

I  did notice that detail in your paper. However, When I tried to plot both the ground truth $`v^r`$ and $`v^b`$, neither matched your Fig.3. Here's what I did:
- Calculate the ground truth $` v^b `$ from the given global frame $` v^n `$ as
```math
 v^b = C^b_n v_b^n 
```
- Calculate the ground truth $` v^r `$ as 
```math
 v^r = C^r_b v^b 
``` 
Note that the calibration file provides $` C^b_r `$.

Here are the results

![fig_3_compare](https://github.com/user-attachments/assets/783427d3-7064-4249-960a-af7c67e97cc9)

The first column represents X-Y-Z axes of $` v^b `$ and second column represents $` v^r `$.
As I observed, the Z axis in your Fig.3 is more similar to the Z axis of $` v^n `$ rather than $` v^r `$ or $` v^b `$.

**Comment by StironjaVlaho** [comment]
We applied the same equations as you wrote. 

Firstly, we read the given ground truth velocity values (from the .txt file) in the world frame. We then transformed them to the body frame using the same equations you mentioned, and finally to the radar frame. In the attachment, I have included our ground truth velocity visualization in the world, body, and radar frames for "mocap dark fast" sequence.

$v^n$
![world](https://github.com/user-attachments/assets/26fcfa18-283f-4314-b34b-a9711c7732eb)

$v^b$
![v_b](https://github.com/user-attachments/assets/121489a3-fe1f-42c2-9f25-207bfb891e66)

$v^r$
![vr](https://github.com/user-attachments/assets/53206a8d-0228-4546-b0c4-2774f77a5255)


Did you use the given ground truth velocity values, or did you calculate them using the ground truth pose information?

**Comment by hoangvietdo** [comment]
I directly used the given ground truth velocity. Here are the more detailed steps:

- I loaded the CSV file, extracted the velocity data (index 9:11 using MATLAB indexing), extracted the quaternion, and converted it into Euler angles.
- Transforming the ground truth (velocity and Euler angles) from the FLU frame to the FRD frame to match the IMU coordinates (X=X, Y=-Y, Z=-Z).
- I proceeded with the other steps as I previously explained.

I tried to keep everything original by not transforming the coordinates, but I still couldn't obtain your results :(
